observer/tests/test_extraction.py

```
@@ -0,0 +1,1108 @@
+
+    @patch("observer.extraction.run_agent")
+    def test_batch_limit(self, mock_run, db, tmp_path):
+        """Batch limit restricts how many events are fetched."""
+        mock_run.side_effect = [
+            _mock_agent_response('{"artifacts": [], "edges": []}'),
+            _mock_agent_response('{"summary": "s"}'),
+        ]
+
+        events_data = []
+        # 5 agent text events — batch limit of 2 should only process 2
+        for i in range(5):
```

Ruff (0.15.0)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)